src/notifications/permissions.ts

```diff
@@ -9,14 +10,14 @@ import * as i18n from '@/languages';
 
 export const getPermissionStatus = () => messaging().hasPermission();
 
-export const requestPermission = (): Promise<boolean> => {
-  return new Promise((resolve, reject) => {
-    requestNotifications(['alert', 'sound', 'badge'])
-      .then(({ status }) => {
-        resolve(status === 'granted');
-      })
-      .catch(e => reject(e));
-  });
+export const requestNotificationPermission = async (): Promise<PermissionStatus> => {
+  const notificationSetting = await requestNotifications(['alert', 'sound', 'badge']);
+  const { status } = notificationSetting;
+  const enabled = status === RESULTS.GRANTED || status === RESULTS.LIMITED;
```

Comment:
```suggestion
  const enabled = [RESULTS.GRANTED, RESULTS.LIMITED].includes(status)
```
I also see this being used in a few different places, maybe break this out into a function so we can just define it once?